packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -242,33 +249,124 @@ export interface IContainerRuntimeMessageCompatDetails {
  * @param compatBehavior - Typed redundantly with CompatModeBehavior to ensure handling is added when updating that type
  */
 function compatBehaviorAllowsMessageType(
-	_unknownContainerRuntimeMessageType: never,
+	_unknownContainerRuntimeMessageType: UnknownContainerMessageType,
 	compatBehavior: "Ignore" | "FailToProcess" | undefined,
 ): boolean {
 	// undefined defaults to same behavior as "FailToProcess"
 	return compatBehavior === "Ignore";
 }
 
 /**
- * The unpacked runtime message / details to be handled or dispatched by the ContainerRuntime
+ * The unpacked runtime message / details to be handled or dispatched by the ContainerRuntime.
+ * Message type are differentiated via a `type` string and contain different contents depending on their type.
  *
  * IMPORTANT: when creating one to be serialized, set the properties in the order they appear here.
  * This way stringified values can be compared.
  */
-export interface ContainerRuntimeMessage {
+export interface ContainerRuntimeMessage<TType, TContents> {
 	/** Type of the op, within the ContainerRuntime's domain */
-	type: ContainerMessageType;
+	type: TType;
 	/** Domain-specific contents, interpreted according to the type */
-	contents: any;
+	contents: TContents;
 	/** Info describing how to handle this op in case the type is unrecognized (default: fail to process) */
 	compatDetails?: IContainerRuntimeMessageCompatDetails;
 }
```

Comment:
This is from the past, but is it expected to exist or not exist on any of these specific messages?
(I am have the impression that it is a future proofing aspect but not expected for the already known cases. Haven't read the code.)

Reply:
That's correct, no current code puts `compatDetails` on.  In coming sprints I'll add a case that does though.  Down the road I'll probably remove it once enough time / versions have passed.

Reply:
Sure. So in that future thing you will do (correct me if I am wrong), there will be a new ContainerRuntimeMessageType entry. The outbound type must provide `compatDetails` and the inbound case could have it but at most it would be optional (since you want to remove it at some point).
So here and now, that should only appear on the Unknown case.